src/mon/OSDMonitor.cc

```diff
@@ -876,7 +877,10 @@ void OSDMonitor::update_from_paxos(bool *need_bootstrap)
 
   share_map_with_random_osd();
   update_logger();
-
+  unsigned cur_osd_num = osdmap.get_num_osds();
+  if(last_osd_num != 0 && last_osd_num != cur_osd_num){
```

Comment:
these criteria are not quite right. as an OSD could be removed from the crush map after it sent a failure report to the monitor. and another OSD is added afterwards before we compare the old and new `osdmap.get_num_osds()`.